aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -27,37 +27,192 @@ Kestrel supports the following features:
+
+```csharp
+[RequestSizeLimit(100000000)]
+public IActionResult MyActionMethod()
+```
+
+You can also override the setting on a specific request as shown here:
+
+[!code-csharp[](kestrel/sample2/Startup.cs?name=snippet_Limits&highlight=3-4)]
+ 
+An exception is thrown if you try to configure the limit on a request after the application has started reading the request. There's an `IsReadOnly` property that tells you if the request body is in read-only state, meaning it's too late to configure the limit.
+
+**Minimum request body data rate**
+
+Here's an example that shows how to configure the minimum request rate:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=6-7)]
+
+You can configure the rate per request:
+
+[!code-csharp[](kestrel/sample2/Startup.cs?name=snippet_Limits&highlight=5-6)]
+
+Kestrel checks every second if data is coming in at the specified rate in bytes/second. If the rate drops below the minimum, the connection is timed out. The grace period is the amount of time that Kestrel gives the client to increase its send rate up to the minimum; the rate is not checked during that time. The grace period helps avoid dropping connections that are initially sending data at a slow rate due to TCP slow-start.
+
+For information about other Kestrel options, see [KestrelServerOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerOptions.cs) and [KestrelServerLimits](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerLimits.cs). 
+
+---
+
+### Endpoint configuration
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 By default ASP.NET Core binds to `http://localhost:5000`. You can configure URL prefixes and ports for Kestrel to listen on by using the `UseUrls` extension method, the `urls` command-line argument, or the ASP.NET Core configuration system. For more information about these methods, see [Hosting](../../fundamentals/hosting.md). For information about how URL binding works when you use IIS as a reverse proxy, see [ASP.NET Core Module](aspnet-core-module.md). 
 
-URL prefixes for Kestrel can be in any of the following formats. 
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+By default ASP.NET Core binds to `http://localhost:5000`. You configure URL prefixes and ports for Kestrel to listen on by calling `Listen` or `ListenUnixSocket` methods on `KestrelServerOptions`. (`UseUrls` and the `urls` command-line argument also work but have the limitations noted [later in this article](#useurls-limitations).)
+
+**Bind to a TCP socket**
+
+The `Listen` method binds to a TCP socket, and an options lambda lets you configure an SSL certificate:
```

Comment:
I guess calling UseKestrel after calling WebHost.CreateDefaultBuilder would also be a good way to show this.